### ISSUE TITLE
remove 'trace' option from message log UI

### ISSUE THREAD

**Issue by ktamail666** [transmission/transmission#4233]
### What is the issue?

Reproduce:
Start GTK 3 client.
Help -> Message Log. Select level Trace. Start to download any kind of torrent. Wait about 1 or 2 mins and you'll got not responding window.

OS: Debian Bookworm x86_64. Custom compile with default flags, clang14 from "main" branch(3ecf0dfccd627c1a137bd6b5103f9190e0d96af6).
Debug/Release also not responding.

First run gdb backtrace:
```
(gdb) bt
#0  0x00007ffff70ae155 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#1  0x00007ffff70ae7a1 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#2  0x00007ffff70ae436 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#3  0x00007ffff70ae29f in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#4  0x00007ffff70b0a18 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#5  0x00007ffff70b0ddc in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#6  0x00007ffff71766e0 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff7165699 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#12 0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff7161f5d in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#17 0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#18 0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#19 0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#20 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#21 0x00007ffff704672c in gtk_list_store_set_value () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#22 0x00005555556849bf in Gtk::TreeRow::set_value<unsigned int>(Gtk::TreeModelColumn<unsigned int> const&, unsigned int const&) const (this=0x7fffffffd280, column=..., data=@0x555555b87e88: 19962) at /usr/include/gtkmm-3.0/gtkmm/treeiter.h:580
#23 0x00005555556843d8 in Gtk::TreeValueProxy<unsigned int>::operator=(unsigned int const&) (this=0x7fffffffd1f0, data=@0x555555b87e88: 19962) at /usr/include/gtkmm-3.0/gtkmm/treeiter.h:552
#24 0x00005555556ee63f in (anonymous namespace)::addMessages(Glib::RefPtr<Gtk::ListStore> const&, tr_log_message*) (store=..., head=0x555557b755e0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:366
#25 0x00005555556ee2e4 in MessageLogWindow::Impl::onRefresh() (this=0x5555564b4150) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:397
#26 0x00005555556f9412 in sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>::operator()() const (this=0x555556f375c8) at /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
#27 0x00005555556f9389 in sigc::adaptor_functor<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl> >::operator()() const (this=0x555556f375c0) at /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
#28 0x00005555556f9361 in sigc::internal::slot_call0<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>, bool>::call_it(sigc::internal::slot_rep*) (rep=0x555556f37590) at /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
#29 0x00007ffff66a5fb2 in  () at /lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#30 0x00007ffff650719a in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#31 0x00007ffff650667f in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#32 0x00007ffff6506a38 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#33 0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#34 0x00007ffff67b353d in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#35 0x00005555556c78ce in main(int, char**) (argc=1, argv=0x7fffffffe008) at /home/xxx/dev/transmission/gtk/main.cc:127
(gdb) f 24
#24 0x00005555556ee63f in (anonymous namespace)::addMessages (store=..., head=0x555557b755e0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:366
366	        row[message_log_cols.sequence] = ++sequence;
(gdb) l
361	        auto row_it = store->prepend();
362	        auto& row = *row_it;
363	        row[message_log_cols.tr_msg] = &message;
364	        row[message_log_cols.name] = name;
365	        row[message_log_cols.message] = message.message;
366	        row[message_log_cols.sequence] = ++sequence;
367	
368	        /* if it's an error message, dump it to the terminal too */
369	        if (message.level == TR_LOG_ERROR)
370	        {
(gdb) info locals
message = @0x555557b75570: {level = TR_LOG_TRACE, file = "/home/xxx/dev/transmission/libtransmission/bandwidth.cc", line = 187, when = 1669274052, name = Python Exception <class 'gdb.error'>: There is no member named _M_p.
, message = Python Exception <class 'gdb.error'>: There is no member named _M_p.
, next = 0x555557b755e0}
name = 0x555557b63e80 "bandwidth.cc:187"
row_it = {<Gtk::TreeIterBase> = {<No data fields>}, model_ = 0x555556c87688, is_end_ = false}
row = @0x7fffffffd280: {<Gtk::TreeIter> = {<Gtk::TreeIterBase> = {<No data fields>}, model_ = 0x555556c87688, is_end_ = false}, <No data fields>}
sequence = 19962
default_name = <incomplete type>
```

Second try backtrace:
```
(gdb) thread apply all bt

Thread 19 (Thread 0x7fffedc706c0 (LWP 124683) "transmission-gt"):
#0  __futex_abstimed_wait_common64 (private=0, cancel=true, abstime=0x0, op=393, expected=0, futex_word=0x5555565f4c20) at ./nptl/futex-internal.c:57
#1  __futex_abstimed_wait_common (futex_word=futex_word@entry=0x5555565f4c20, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0, cancel=cancel@entry=true) at ./nptl/futex-internal.c:87
#2  0x00007ffff5bc5d9b in __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x5555565f4c20, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ./nptl/futex-internal.c:139
#3  0x00007ffff5bc83f8 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x5555565f4bd0, cond=0x5555565f4bf8) at ./nptl/pthread_cond_wait.c:503
#4  ___pthread_cond_wait (cond=0x5555565f4bf8, mutex=0x5555565f4bd0) at ./nptl/pthread_cond_wait.c:618
#5  0x0000555555854575 in std::condition_variable::wait<tr_web::Impl::curlThreadFunc(tr_web::Impl*)::{lambda()#1}>(std::unique_lock<std::mutex>&, tr_web::Impl::curlThreadFunc(tr_web::Impl*)::{lambda()#1}) (this=0x5555565f4bf8, __lock=..., __p=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/condition_variable:102
#6  0x0000555555852f03 in tr_web::Impl::curlThreadFunc(tr_web::Impl*) (impl=0x5555565f4b40) at /home/xxx/dev/transmission/libtransmission/web.cc:497
#7  0x0000555555854285 in std::__invoke_impl<void, void (*)(tr_web::Impl*), tr_web::Impl*>(std::__invoke_other, void (*&&)(tr_web::Impl*), tr_web::Impl*&&) (__f=@0x555556cf4e50: 0x555555852db0 <tr_web::Impl::curlThreadFunc(tr_web::Impl*)>, __args=@0x555556cf4e48: 0x5555565f4b40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:61
#8  0x00005555558541d2 in std::__invoke<void (*)(tr_web::Impl*), tr_web::Impl*>(void (*&&)(tr_web::Impl*), tr_web::Impl*&&) (__fn=@0x555556cf4e50: 0x555555852db0 <tr_web::Impl::curlThreadFunc(tr_web::Impl*)>, __args=@0x555556cf4e48: 0x5555565f4b40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:96
#9  0x0000555555854192 in std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) (this=0x555556cf4e48) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:252
#10 0x0000555555854145 in std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> >::operator()() (this=0x555556cf4e48) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:259
#11 0x0000555555854019 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> > >::_M_run() (this=0x555556cf4e40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:210
#12 0x00007ffff58d44a3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#13 0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#14 0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
Warning: the current language does not match this frame.

Thread 18 (Thread 0x7fffecb146c0 (LWP 124682) "transmission-gt"):
#0  0x00007ffff5c48cb6 in epoll_wait (epfd=16, events=0x555556436f00, maxevents=32, timeout=40) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
#1  0x00007ffff63d7814 in  () at /lib/x86_64-linux-gnu/libevent-2.1.so.7
#2  0x00007ffff63cd8a1 in event_base_loop () at /lib/x86_64-linux-gnu/libevent-2.1.so.7
#3  0x0000555555941d5a in tr_session_thread_impl::sessionThreadFunc(event_base*) (this=0x55555658dad0, evbase=0x55555662c580) at /home/xxx/dev/transmission/libtransmission/session-thread.cc:244
#4  0x00005555559430e9 in std::__invoke_impl<void, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*>(std::__invoke_memfun_deref, void (tr_session_thread_impl::*&&)(event_base*), tr_session_thread_impl*&&, event_base*&&) (__f=@0x555556807248: (void (tr_session_thread_impl::*)(tr_session_thread_impl * const, event_base *)) 0x555555941c80 <tr_session_thread_impl::sessionThreadFunc(event_base*)>, __t=@0x555556807240: 0x55555658dad0, __args=@0x555556807238: 0x55555662c580) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:74
#5  0x0000555555942fb7 in std::__invoke<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*>(void (tr_session_thread_impl::*&&)(event_base*), tr_session_thread_impl*&&, event_base*&&) (__fn=@0x555556807248: (void (tr_session_thread_impl::*)(tr_session_thread_impl * const, event_base *)) 0x555555941c80 <tr_session_thread_impl::sessionThreadFunc(event_base*)>, __args=@0x555556807240: 0x55555658dad0, __args=@0x555556807238: 0x55555662c580) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:96
#6  0x0000555555942f5b in std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >::_M_invoke<0ul, 1ul, 2ul>(std::_Index_tuple<0ul, 1ul, 2ul>) (this=0x555556807238) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:252
#7  0x0000555555942ef5 in std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >::operator()() (this=0x555556807238) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:259
#8  0x0000555555942cf9 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> > >::_M_run() (this=0x555556807230) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:210
#9  0x00007ffff58d44a3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#10 0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#11 0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 11 (Thread 0x7ffff01fe6c0 (LWP 124675) "dconf worker"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x555556577e50, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007fffee47d3bd in  () at /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 3 (Thread 0x7ffff35bd6c0 (LWP 124666) "gdbus"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x5555569cee80, nfds=4, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506cef in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff67ee296 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 2 (Thread 0x7ffff3dbe6c0 (LWP 124665) "gmain"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x5555569b1c10, nfds=2, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff6506b11 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 1 (Thread 0x7ffff413f0c0 (LWP 124664) "transmission-gt"):
#0  0x00007ffff65216f3 in g_sequence_foreach_range () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007ffff716191e in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#2  0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#3  0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#4  0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff7046d94 in gtk_list_store_insert () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff7b7d48a in Gtk::ListStore::prepend() () at /lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#8  0x00005555556ee45b in (anonymous namespace)::addMessages(Glib::RefPtr<Gtk::ListStore> const&, tr_log_message*) (store=..., head=0x555557b202a0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:361
#9  0x00005555556ee2e4 in MessageLogWindow::Impl::onRefresh() (this=0x555556e343f0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:397
#10 0x00005555556f9412 in sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>::operator()() const (this=0x555556f1d028) at /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
#11 0x00005555556f9389 in sigc::adaptor_functor<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl> >::operator()() const (this=0x555556f1d020) at /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
#12 0x00005555556f9361 in sigc::internal::slot_call0<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>, bool>::call_it(sigc::internal::slot_rep*) (rep=0x555556f1cff0) at /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
#13 0x00007ffff66a5fb2 in  () at /lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#14 0x00007ffff650719a in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#15 0x00007ffff650667f in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#16 0x00007ffff6506a38 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#17 0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007ffff67b353d in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#19 0x00005555556c78ce in main(int, char**) (argc=1, argv=0x7fffffffe008) at /home/xxx/dev/transmission/gtk/main.cc:127
```

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

https://github.com/transmission/transmission/commit/3ecf0dfccd627c1a137bd6b5103f9190e0d96af6

**Comment by ckerr** [comment]
Trace generates _so much_ data that I think it just is overwhelming in a UI app.

Maybe it should not be an option in the GUI apps

**Comment by ckerr** [comment]
@GaryElshaw did you mention something like this in the macOS client in the nightlies?

**Comment by ktamail666** [comment]
Thanks for quick feedback. Yes I think trace level is not good idea on GTK client. Debug level works fine. Sorry to bother you.
You can close ticket, I can live without trace levels...

**Comment by GaryElshaw** [comment]
> @GaryElshaw did you mention something like this in the macOS client in the nightlies?

Yes, there is an **enormous** amount of debug and trace data collected in the message log. One of the important points in the macOS client is that the debug data is treated as trace. Although the macOS client has a separate parameter for viewing Trace, it is always purely information from Debug and there is currently no method of controlling the volume of data collected. Raising a separate issue? Is this information of any value to maintainers?

**Comment by ckerr** [comment]
> One of the important points in the macOS client is that the debug data is treated as trace. Although the macOS client has a separate parameter for viewing Trace, it is always purely information from Debug and there is currently no method of controlling the volume of data collected. 

I'm not understanding this; could you rephrase and have another try?

The different verbosity levels are:

- critical (errors that prevent transmission from running)
- error (errors that could prevent a single torrent from running but not block the whole system)
- warn (smaller errors that don't prevent running, e.g. preallocate failed)
- info (user-visible info, e.g. "torrent completed" or "runing script")
- debug
- trace (high-volume debug)

So when you say "debug data is always treated as trace", I don't understand what that means?

**Comment by Coeur** [comment]
We could somehow reduce the volume if there wasn't a constant UTP error...
![Capture d’écran 2022-11-25 à 00 53 46](https://user-images.githubusercontent.com/839992/203835490-eccc083b-8760-42e5-b8c1-cb13fa43e902.png)

**Comment by GaryElshaw** [comment]
> So when you say "debug data is always treated as trace", I don't understand what that means?

Sorry about that, i'm trying to explain something that might just be my own logic.

These are the view options in the macOS client:

![SCR-20221125-sw6](https://user-images.githubusercontent.com/69029666/203928506-69aa90f1-29b3-406f-be37-83a88a1857c6.png)

Yep, these are all of the parameters discussed. My question, really, is about three things Which is why it's probably confusing). One is about the function and logic of what is being recorded; the second part is about user control of *what* and how much is being recorded; thirdly, how useful is this information for maintainers?

At the moment, the 'Trace' option includes a purple graphic that should theoretically illustrate what is Trace data being logged. That purple graphic is not used or represented in an identifiable way like the other options are in 'Message Log'.

![SCR-20221125-stc](https://user-images.githubusercontent.com/69029666/203931206-09729fcc-114b-4084-b787-b9be216ac32c.png)

Info, Debug, Warning, Error etc are all contained in Trace, but there is no 'Trace' per se (a purple icon marking that as separate information, unlike the green, red, orange etc).

My argument is that Trace should be removed if there is no identifiable information that is 'Trace'. It's just information contained elsewhere in the Message Log ui.

Secondly. It would be great if users had some control over all of the logging functionality. Ideally, i would love to see an ios style on/off slide widget to turn off logging for each of the options.

![SCR-20221125-tet](https://user-images.githubusercontent.com/69029666/203933684-8ce02ef4-a568-4f65-b3c1-d76f142f5c3a.png)

Thirdly: My gut reaction when i see a lot of error and warning messages is to report them to the project, because, in theory, they are issues that need to be resolved. I know error warnings are often NOT because of Transmission but peers and tracker-related, but trying to differentiate is, well, _difficult_.

![SCR-20221125-sur](https://user-images.githubusercontent.com/69029666/203934813-e81179b2-9f75-4173-ac6e-1edb42f57681.png)

**Comment by Coeur** [comment]
Trying to interpret the code that leads to "utp_on_error -- errcode is 2", it seems to theoretically happen after four absences of responses:
https://github.com/transmission/libutp/blob/bf695bdfb047cdca9710ea9cffc4018669cf9548/utp_internal.cpp#L1187-L1188

...with a timer set to fire every 50-150ms:
https://github.com/transmission/transmission/blob/6b861806a653c6490c247819f7959ca39f54e752/libtransmission/tr-utp.cc#L160-L163

So total it seems equivalent to a peer not responding in less than 0.2s-0.6s.

Note that I'm in China, and the Great Firewall forces most connections crossing the borders to be tampered with a ~1s delay, so I'm not in the best conditions.

**Comment by ckerr** [comment]
> So total it seems equivalent to a peer not connecting in less than 0.1s-0.3s.

Hm, is this even logging, even as a trace message? I'm not sure that's something users or devs would want to know about

**Comment by Coeur** [comment]
@ckerr Logged here when errcode is 2:
https://github.com/transmission/transmission/blob/6b861806a653c6490c247819f7959ca39f54e752/libtransmission/peer-io.cc#L390

@GaryElshaw trace seems to be used in code: search for occurrences of `tr_logAddTrace(` and `logtrace(`.

**Comment by GaryElshaw** [comment]
Interesting. I've never seen anything tagged as 'Trace', have you seen it Antoine?

**Comment by Coeur** [comment]
Ah, I get it why we never see the trace messages: `tr_logSetLevel` is called once with the default settings (DEBUG) and is never modified: the macOS UI will simply filter the given messages for various ranges, but since we didn't request the TRACE ones, we never get them.

**Comment by Coeur** [comment]
I just tried enabling TR_LOG_TRACE... and the app is freezing dead: the amount of messages is too high. So I guess yes, we should remove the violet dot option from the UI.

**Comment by mikedld** [comment]
> Hm, is this even logging, even as a trace message? I'm not sure that's something users or devs would want to know about

Deja vu: https://github.com/transmission/transmission/issues/2996#issuecomment-1164703109

**Comment by Coeur** [comment]
@mikedld that's a different issue, which I'm also addressing, but separately in #4237

**Comment by ckerr** [comment]
> > Hm, is this even logging, even as a trace message? I'm not sure that's something users or devs would want to know about
> 
> Deja vu: [#2996 (comment)](https://github.com/transmission/transmission/issues/2996#issuecomment-1164703109)

[And that's why you always open separate issues for separate topics](https://www.youtube.com/watch?v=IJw2lFWJ4aY) :mechanical_arm:
